prac_07/project_management.py

```diff
@@ -0,0 +1,199 @@
+"""
+Estimate time: 1h
+Actual time: 7h
+"""
+import datetime
+from prac_07.project import Project
+from operator import attrgetter
+
+FILENAME = "projects.txt"
+MENU = ("- (L)oad project\n- (S)ave projects\n- (D)isplay projects\n- (F)ilter projects by date\n- (A)dd new project\n"
+        "- (U)pdate project\n- (Q)uit")
+MINIMUM_PROJECT_CHOICE = 0
+MINIMUM_PERCENTAGE = 0
+MAXIMUM_PERCENTAGE = 100
+MINIMUM_PRIORITY = 0
+MAXIMUM_PRIORITY = 20
+MINIMUM_COST_ESTIMATE = 0
+MAXIMUM_COST_ESTIMATE = 999999
+
+
+def main():
+    """Play the program according to the menu."""
+    data = []
+    load_data(data, FILENAME)
+    print("Welcome to Pythonic Project Management")
+    print(f"Loaded {len(data)} projects from {FILENAME}")
+    print(MENU)
+    choice = input(">>> ").upper()
+    while choice != "Q":
+        if choice == "L":
+            load_project(data)
+        elif choice == "S":
+            save_project(data)
+        elif choice == "D":
+            display_projects(data)
+        elif choice == "F":
+            filter_project(data)
+        elif choice == "A":
+            add_new_project(data)
+        elif choice == "U":
+            update_project(data)
+        else:
+            print("Invalid choice")
+        print(MENU)
+        choice = input(">>> ").upper()
+    save_data(data, FILENAME)
+    print("Thank you for using custom-built project management software.")
+
+
+def save_project(data):
+    """Save project."""
+    new_saving_filename = input("New saving filename: ")
+    save_data(data, new_saving_filename)
+
+
+def load_project(data):
+    """Load project."""
+    new_loading_filename = input("New filename:")
+    load_data(data, new_loading_filename)
+
+
+def filter_project(data):
+    """Filter project by date."""
+    filtered_projects = []
```

Comment:
This is a good name, however it doesn't match the name previously used for the projects list. Here you call them projects, earlier you use data. It's easier to read code when names are consistent.